tests/api/test_remote_engine.py

```diff
@@ -120,18 +124,21 @@ def test_run_async(self, prog):
         engine = RemoteEngine("X8_01")
         job = engine.run_async(prog, shots=10)
 
-        assert job.status == JobStatus.OPEN.value
+        # job.status calls job.finished, incrementing the request counter
+        assert job.status == "open"
 
-        for _ in range(REQUESTS_BEFORE_COMPLETED):
-            job.refresh()
+        for _ in range(REQUESTS_BEFORE_COMPLETED - 1):
+            assert job.finished == False
+        assert job.finished == True
```

Comment:
It is more Pythonic to compare Booleans using `is`:
```suggestion
            assert job.finished is False
        assert job.finished is True
```